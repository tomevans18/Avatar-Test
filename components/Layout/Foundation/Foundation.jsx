import PropTypes from 'prop-types';
import React, { Fragment } from 'react';

import Header from '../../Header';
import Content from '../../Content';
import Footer from '../../Footer';

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]).isRequired,
};

const Foundation = props => {
  const { children } = props;
  return (
    <Fragment>
      <Header mainId="#mainContent" />
      <Content id="mainContent" tabIndex="-1">
        {children}
      </Content>
      <Footer />
    </Fragment>
  );
};

Foundation.propTypes = propTypes;

export default Foundation;
