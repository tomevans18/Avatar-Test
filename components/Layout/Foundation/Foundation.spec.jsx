import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Foundation from './Foundation';

describe('Layout - Foundation', () => {
  it('should render as expected', () => {
    const wrapper = shallow(
      <Foundation>
        <div>TEST CONTENT</div>
      </Foundation>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
