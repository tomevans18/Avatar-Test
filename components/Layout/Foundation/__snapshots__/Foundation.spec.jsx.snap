// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout - Foundation should render as expected 1`] = `
<Fragment>
  <Header
    mainId="#mainContent"
  />
  <Content
    id="mainContent"
    tabIndex="-1"
  >
    <div>
      TEST CONTENT
    </div>
  </Content>
  <Footer />
</Fragment>
`;
